tsl/src/nodes/decompress_chunk/vector_quals.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+#pragma once
+
+#include <postgres.h>
+#include <executor/tuptable.h>
+#include <nodes/primnodes.h>
+
+#include "vector_predicates.h"
+#include <compression/arrow_c_data_interface.h>
+
+/*
+ * VectorQualInfo provides planner time information for extracting
+ * vectorizable quals from regular quals.
+ */
+typedef struct VectorQualInfo
+{
+	/* The range-table index of the relation to compute vectorized quals
+	 * for */
+	Index rti;
+
+	/*
+	 * Interface function to be provided by scan node.
+	 *
+	 * Given an attribute/column number, determine whether the column is
+	 * vectorizable.
+	 *
+	 * Scan-node specific data can be provided by embedding this struct in a
+	 * larger one.
+	 */
+	bool (*is_vector_column)(const struct VectorQualInfo *vqinfo, AttrNumber attno);
```

Comment:
Can this be a bool array? I'd prefer having plain data as an interface, simpler control flow is good for reading and debugging the code.

Reply:
Changed.